### PR TITLE
Let a page AI agent default its sessions into a drive Environment

### PR DESCRIPTION
**Pull request by 2witstudios**
## Summary
- Adds `pages.defaultEnvId` so an AI_CHAT agent's Settings screen (next to the existing Sandbox toggle) can assign one of the drive's persistent Environments as that agent's default.
- This is a **default, not a binding**: the spawn palette pre-highlights the assigned environment (with a "Default" badge) but ephemeral sandboxes and other environments stay fully selectable, and the server (`POST /api/agent-workspaces`) falls back to the agent's default only when the caller omits `envId` entirely — so any caller, not just the web UI, gets consistent behavior.
- `GET`/`PATCH /api/pages/[pageId]/agent-config` gained validation (string-or-null, non-empty, env exists, same-drive only) and now also echo `sandboxEnabled`/`toolExposureMode` back in the PATCH response (a pre-existing gap fixed while touching this route).

## Test plan
- [x] `bun run typecheck` (monorepo-wide) passes
- [x] `bun run lint` passes
- [x] `bun run build` (web) passes
- [x] New unit tests added for the agent-config route (validation: type, empty string, unknown env, cross-drive rejection, same-drive success, clearing to `null`, response echo) — all pass
- [x] New unit tests added for the agent-workspaces spawn route (server-side default fallback, explicit `envId` takes precedence) — all pass
- [x] Mutation-tested the three new validation/fallback branches (same-drive check, existence check, server fallback) by breaking each and confirming the corresponding test goes red
- [x] Existing `drive-envs` schema/service test suites pass unaffected by the new FK
- [ ] Manual spawn-flow verification in a running app (not performed in this session — recommend before merge: assign a default env to an agent in Settings, open the spawn palette, confirm the "Default" badge and that other options remain selectable)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01HRWUp5GAEFWNPCkMTzVDob

## Summary by CodeRabbit

* **New Features**
  * Added an Environment selector to agent settings for choosing the default environment for new sessions.
  * The selected default environment is clearly labeled and shown first when starting a session.
  * Default environments are available across agent listings and session creation.

* **Bug Fixes**
  * New agent sessions now automatically use the configured default environment when none is specified.
  * Explicit environment selections continue to take precedence.
  * Invalid, unavailable, or cross-drive environments are rejected, and deleted environments are cleared safely.

### CODE REVIEW

File: apps/web/src/app/api/agent-workspaces/route.ts [hunk 1]
```diff
@@ -393,6 +393,15 @@ export async function POST(request: Request) {
...
     if (denied) return denied;
     agentTitle = agent.title;
+    // Fall back to the agent's own default env when the caller didn't say —
+    // "didn't specify" means "use my default," not "force ephemeral." An
+    // explicit `envId` in the request always wins; this only fires when the
+    // caller sent nothing, so every caller of this route (palette, any
+    // future MCP/API tool) gets the agent's Settings-screen default without
+    // having to know it exists.
+    if (envId === null && agent.defaultEnvId) {
+      envId = agent.defaultEnvId;
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply apps/web/src/app/api/agent-workspaces/route.ts]: Fixed — thanks for catching this. `envId` presence is now tracked separately (`envIdWasProvided`, via `Object.prototype.hasOwnProperty`), so an explicit `null` (the palette's "New sandbox" pick) is distinguished from a truly omitted field. The fallback to the agent's `defaultEnvId` now only fires when the caller sent nothing at all. Also added a regression test asserting explicit `null` stays ephemeral even when the target agent has a default. See commit ccd8efc7d.

File: packages/db/src/schema/core.ts [hunk 2]
```diff
@@ -60,6 +61,7 @@ export const pages = pgTable('pages', {
...
   sandboxEnabled: boolean('sandboxEnabled').default(false).notNull(), // AI_CHAT agents: whether the sandbox tool families (bash/files, git+gh, sessions/shells) are offered to this agent. Provisioning stays lazy and automatic on first use — this is the settings switch, not a provision button. Successor to the old pages.terminalAccess (dropped in 0234, phase 8's teardown of the Machines model — never pages.machineAccess, which never existed in this package).
+  defaultEnvId: text('defaultEnvId').references((): AnyPgColumn => driveEnvs.id, { onDelete: 'set null' }), // AI_CHAT agents: the drive Environment (drive_envs) pre-selected when spawning a new session for this agent. A DEFAULT, not a binding — spawn time (useSpawnSession / POST /api/agent-workspaces) still allows overriding to ephemeral or another env. NULL = no default (ephemeral). Not coupled at the DB level to sandboxEnabled — inert data when the switch is off, enforced only in the settings UI. `onDelete: 'set null'` so deleting the env falls the agent back to ephemeral rather than touching the page row.
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply packages/db/src/schema/core.ts]: Fixed — cross-drive moves (both the root page and every cascaded descendant) now clear `defaultEnvId` when the page actually changes drives, since the env it pointed to belongs to the drive being left. Added tests covering the root move, the cascade batch, and the same-drive-reparent case (which must NOT clear it). See commit ccd8efc7d.

File: apps/web/src/app/api/agent-workspaces/route.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (2witstudios) [reply apps/web/src/app/api/agent-workspaces/route.ts]: Fixed — the spawn-time fallback (and the same fallback now also applied to the `firstThing: 'claim'` branch and to `resolveWorkerPlacement`'s "new workspace" path, both of which had the same gap) now checks the agent's live `sandboxEnabled` in addition to `defaultEnvId`, so a retained default stays inert while Sandbox is off, matching the settings-route's documented contract. See commits ccd8efc7d and 06ee24937.

File: apps/web/src/components/agents/useSpawnSession.tsx [hunk 3]
```diff
@@ -723,7 +735,19 @@ function SpawnSessionPalette({
                 <span className="truncate">New sandbox</span>
                 <span className="ml-auto shrink-0 text-xs text-muted-foreground">Ephemeral</span>
               </CommandItem>
-              {envs.map((env) => (
+              {defaultEnv && (
+                <CommandItem
+                  key={defaultEnv.id}
+                  value={`${defaultEnv.id}-in ${defaultEnv.name}`}
+                  disabled={spawning}
+                  onSelect={() => onPickEnv(defaultEnv.id)}
```
[the remark this answers is not in the document]
Comment (2witstudios) [reply apps/web/src/components/agents/useSpawnSession.tsx]: Fixed — the configured default environment now renders BEFORE "New sandbox" in the env-picker list when one exists, since cmdk's uncontrolled initial highlight tracks DOM order (not array order or a controlled `value` this dialog doesn't thread through). "New sandbox" and every other environment remain fully reachable and keep their prior relative order otherwise. See commit ccd8efc7d.
